lib/chef/mixin/api_version_request_handling.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Author:: Tyler Cloke (tyler@chef.io)
+# Copyright:: Copyright 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  module ApiVersionRequestHandling
+    # takes in an http exception, and a min and max supported API version and
+    # handles all the versioning cases
+    #
+    # it will return false if there was a non-versioning related error
+    # or the server and the client are not compatible
+    #
+    # if the server does not support versioning, then it will return true, and you
+    # can assume API v0 is safe to send
+    def handle_version_http_exception(exception, min_client_supported_version, max_client_supported_version)
```

Comment:
Maybe could be named `should_retry_with_lower_api_version` or something like that?